### ISSUE TITLE
WebappContext.setLoadOnStartup does not work as advertised (bug?) (servlet is created twice)

### ISSUE THREAD

**Issue by zolyfarkas** [eclipse-ee4j/glassfish-grizzly#2034]
setting loadOnStartup will create the servlet on startup at:

https://github.com/eclipse-ee4j/grizzly/blob/30609402b13c7c965ceb5df69a69ccd58d8c85c5/modules/http-servlet/src/main/java/org/glassfish/grizzly/servlet/WebappContext.java#L1883

however, this instance is not being set into the holder at:

https://github.com/eclipse-ee4j/grizzly/blob/30609402b13c7c965ceb5df69a69ccd58d8c85c5/modules/http-servlet/src/main/java/org/glassfish/grizzly/servlet/WebappContext.java#L1891

resulting in the servlet being instantiated again on first request.

as a workaround I am using:

```
package org.glassfish.grizzly.servlet;

import javax.servlet.Servlet;

public final class FixedWebappContext extends WebappContext {

  public FixedWebappContext(String displayName) {
    super(displayName);
  }

  public FixedWebappContext(String displayName, String contextPath) {
    super(displayName, contextPath);
  }

  public FixedWebappContext(String displayName, String contextPath, String basePath) {
    super(displayName, contextPath, basePath);
  }

  @Override
  protected Servlet createServletInstance(ServletRegistration registration) throws Exception {
    Servlet srvlet = super.createServletInstance(registration);
    registration.servlet = srvlet;
    return srvlet;
  }
}
```

**Comment by svendiedrichsen** [comment]
@zolyfarkas Would you mind verifying the fix #2043?

**Comment by zolyfarkas** [comment]
Will take a look, wish these spam filters would not be so hungry for my github notifications...
